engine/src/main/java/com/sandboni/core/engine/sta/connector/TestSuiteConnector.java

```diff
@@ -0,0 +1,45 @@
+package com.sandboni.core.engine.sta.connector;
+
+import com.sandboni.core.engine.sta.Context;
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import com.sandboni.core.engine.utils.StringUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.sandboni.core.engine.sta.graph.vertex.VertexInitTypes.TEST_SUITE_VERTEX;
+
+/**
+ * Connects between each suite test class to it's related methods (test vertices)
+ */
+public class TestSuiteConnector implements Connector {
+
+    @Override
+    public void connect(Context context) {
+        List<Link> testSuiteToTestClassLinks = context.getLinks().filter(l -> (l.getLinkType() == LinkType.TEST_SUITE && !l.getCaller().equals(TEST_SUITE_VERTEX))).collect(Collectors.toList());
```

Comment:
To avoid NPE is a good practice to include the constants on the left side of the compare operation

Reply:
you're right. will be updated